File: app_learn/ios/TestNativeModues.m
//
//  TestNativeModues.m
//  app_learn
//
//  Created by ls-mac on 2017/5/1.
//  Copyright © 2017年 Facebook. All rights reserved.
//

#import "TestNativeModues.h"

@implementation TestNativeModues


RCT_EXPORT_MODULE(TestNativeModues);


RCT_EXPORT_METHOD(test:(NSString *)text
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject){
 
  
  resolve(@"test");
  
}

@end

File: app_learn/android/app/src/main/java/com/app_learn/modules/TestNativeModues.java
package com.app_learn.modules;

import com.facebook.react.bridge.Promise;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;

/**
 * Created by ls-mac on 2017/5/5.
 */

public class TestNativeModues extends ReactContextBaseJavaModule{


    public TestNativeModues(ReactApplicationContext reactContext) {
        super(reactContext);
    }

    @Override
    public String getName() {

        return "TestNativeModues";

    }

    @ReactMethod
    public void test(String text, Promise promise){
        promise.resolve(text);
    }


}
